blogs/secondary_interface/secondary_interface.md

```diff
@@ -0,0 +1,168 @@
+following steps:
+
+1. Using fsNotify Package:
+   Utilize the fsNotify package to be notified when new network namespaces are created.
+   This allows the eBPF agent to keep track of network namespace creation events.
+
+2. Using netlink Package:
+   Employ the netlink package to register when the network interfaces are created or deleted
+   within each network namespace. This will enable the eBPF agent to monitor the interface
+   changes on a per-namespace basis.
+
+3. Attach/Detach eBPF TC Hooks:
+   Add support to the eBPF agent to attach and detach eBPF Traffic Control (TC) hook
+   for network interfaces in non-default network namespaces. This step is crucial for
+   monitoring and controlling network traffic within these network namespaces.
+
+## Configuring SR-IOV objects
+
+1. Install the SR-IOV operator in the environment.
+2. Identify the SR-IOV capable device on the node.
+3. Label the node which has SR-IOV interface with the label
+  `feature.node.kubernetes.io/network-sriov.capable=true`
+4. Create the `SriovNetworkNodePolicy` object
+
+```yaml
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetworkNodePolicy
+metadata:
+  name: mypolicy
+  namespace: openshift-sriov-network-operator
+spec:
+  resourceName: netdeviceresource
+  nodeSelector:
+    feature.node.kubernetes.io/network-sriov.capable: "true"
+  priority: 99
+  numVfs: 50
+  nicSelector:
+    pfNames: ["ens7f0np0#25-49"]
+  deviceType: netdevice
+```
+
+5. Create the `SriovNetwork` object. This will create net-attach-def in
+  the `openshift-sriov-network-operator` namespace.
+
+```yaml
+apiVersion: sriovnetwork.openshift.io/v1
+kind: SriovNetwork
+metadata:
+  name: sriov-test
+  namespace: openshift-sriov-network-operator
+spec:
+  resourceName: netdeviceresource
+  networkNamespace: test
+  ipam: '{ "type": "static", "addresses": [{"address": "192.168.122.71/24"}]}'
+```
+
+6. Create a test pod using the `SRIOVNetwork` object created above and
+   denoted by annotation, `k8s.v1.cni.cncf.io/networks: "sriov-test"`
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: httpd-2
+  namespace: openshift-sriov-network-operator
+  labels:
+    app: sriov
+  annotations:
+    k8s.v1.cni.cncf.io/networks: "sriov-test"
+spec:
+  containers:
+  - name: httpd
+    command: ["sleep", "30d"]
+    image: registry.redhat.io/rhel8/support-tools
+    ports:
+    - containerPort: 8080
+    securityContext:
+      allowPrivilegeEscalation: false
+      seccompProfile:
+        type: RuntimeDefault
+      capabilities:
+        drop:
+        - ALL
+```
+
+## Configuring the network observability operator to work with SR-IOV
+
+1. Deploy the network observability operator.
+
+2. Create the `FollowCollector` object with `privileged` set to `true`.
+
+```yaml
+apiVersion: flows.netobserv.io/v1beta1
+kind: FlowCollector
+metadata:
+  name: cluster
+spec:
+  agent:
+    type: EBPF
+    ebpf:
+      privileged: true
+```
+
```

Comment:
Maybe add a sentence here for transitioning ? Just telling what is going to happen then: netobserv deploys its components, the agents discover all the interfaces and attach their probes, flows start being collected ...